components/hab/src/cli.rs

```diff
@@ -102,6 +104,9 @@ pub fn get() -> App<'static, 'static> {
                 (@arg USER: -u --user +takes_value "Name of the user key")
                 (@arg REMOTE_SUP: --("remote-sup") -r +takes_value
                     "Address to a remote Supervisor's Control Gateway [default: 127.0.0.1:9632]")
+                (arg: arg_cache_key_path("Path to search for encryption keys. \
+                    Default value is hab/cache/keys if root and .hab/cache/keys under the home \
+                    directory otherwise."))
```

Comment:
Since we're using a function to set up the details of the option, and since the help text from "Default value is..." onward is always the same, could we not put that into the function as well? The context specifcity of the first sentence is really nice, though, and I wouldn't want to lose that.

I started thinking down this path because the fact that we're giving relative paths here instead of absolute struck me as odd. Knowing the details of the implementation, I understand why it's that way, but for user-facing documentation it seems like it presumes too much. I'd expect to see `/hab/cache/keys` and `~/.hab/cache/keys` if I were on Linux, for example.

Are users expected to input a relative path?

This may be something for a future PR, but I imagine it would also be nice to show Windows-specific information when using this on Windows, rather than requiring Windows users to mentally translate from Unix-ese.

Reply:
> Since we're using a function to set up the details of the option, and since the help text from "Default value is..." onward is always the same, could we not put that into the function as well? The context specifcity of the first sentence is really nice, though, and I wouldn't want to lose that.

That's what I tried at first. And the answer is that we can't (or I didn't see an obvious way) because the lifetime of the `&str` passed to `help` needs to be `'static`.

> Are users expected to input a relative path?

Both absolute and relative paths are supported. Some of the complication comes from the fact that `hab/cache/keys` isn't necessarily under `/` because `FS_ROOT` is dynamic. I think our whole approach to `hab` vs `.hab` is worth of re-thinking, but I'm trying to keep this particular change narrowly scoped.

Reply:
Ah, that's right, they've got to be static strings!

👍